pkg/resource/deploy/deploytest/resourcemonitor.go

```diff
@@ -338,8 +340,9 @@ func (rm *ResourceMonitor) ReadResource(t tokens.Type, name string, id resource.
 
 	// unmarshal outputs
 	outs, err := plugin.UnmarshalProperties(resp.Properties, plugin.MarshalOptions{
-		KeepUnknowns:  true,
-		KeepResources: true,
+		KeepUnknowns:     true,
+		KeepResources:    true,
+		KeepOutputValues: true,
```

Comment:
Why is it needed?

Reply:
Same as the RegisterResource case, lets us see if these are returned unexpectedly.